trollflow2/__init__.py

```diff
@@ -128,8 +128,11 @@ def save_datasets(job):
         fname_pattern = fmat['fname_pattern']
         filename = compose(os.path.join(fmat['output_dir'], fname_pattern), fmat)
         fmat.pop('format', None)
+        fmat.pop('filename', None)
```

Comment:
Shouldn't it raise an error instead ? there shouldn't be a 'filename' there, right ?

Reply:
Yeah, I was wondering how the filename ends up in the format, but thought it was by design as it was there always for the 2nd and later composites. It is added in the `else` branch, and as I don't understand what `plist_iter` is doing, thought it somehow propagated to the later products.

Reply:
I'm afraid that could be a sign that the format section is shared between multiple products.